src/widgets/notes/visit-notes-form.component.tsx

```diff
@@ -0,0 +1,380 @@
+        results => setSearchResults(results),
+        error => {
+          setError(error);
+          createErrorHandler();
+        }
+      );
+      return () => sub.unsubscribe();
+    } else {
+      setSearchResults(null);
+    }
+  }, [searchTerm]);
+
+  React.useEffect(() => {
+    const ac = new AbortController();
+    fetchCurrentSessionData(ac).then(({ data }) => {
+      const { currentProvider, sessionLocation } = data;
+      if (currentProvider.uuid) {
+        setCurrentSessionProviderUuid(currentProvider.uuid);
+      }
+      if (sessionLocation.uuid) {
+        setCurrentSessionLocationUuid(sessionLocation.uuid);
+      }
+    });
+    return () => ac.abort();
+  }, []);
+
+  React.useEffect(() => {
+    const ac = new AbortController();
+    if (currentSessionProviderUuid) {
+      fetchProviderByUuid(ac, currentSessionProviderUuid).then(({ data }) => {
+        setProviderUuid(data.uuid);
+      });
+    }
+  }, [currentSessionProviderUuid]);
+
+  React.useEffect(() => {
+    const ac = new AbortController();
+    if (currentSessionLocationUuid) {
+      fetchLocationByUuid(ac, currentSessionLocationUuid).then(({ data }) =>
+        setLocationUuid(data.uuid)
+      );
+    }
+  }, [currentSessionLocationUuid]);
+
+  const handleCancel = () => {
+    closeWorkspace();
+  };
+
+  const handleSearchTermChange = debounce(searchTerm => {
+    setSearchTerm(searchTerm);
+  }, searchTimeoutInMs);
+
+  const resetSearch = () => {
+    setSearchTerm("");
+    setSearchResults([]);
+  };
+
+  const handleAddDiagnosis = (diagnosisToAdd: Diagnosis) => {
+    resetSearch();
+    setSelectedDiagnoses(selectedDiagnoses => [
+      ...selectedDiagnoses,
+      diagnosisToAdd
+    ]);
+  };
+
+  const handleRemoveDiagnosis = (diagnosisToRemove: Diagnosis) => {
+    setSelectedDiagnoses(
+      selectedDiagnoses.filter(
+        diagnosis => diagnosis.concept.id !== diagnosisToRemove.concept.id
+      )
+    );
+  };
+
+  const RenderSearchResults: React.FC = () => {
+    if (searchResults.length) {
+      return (
+        <ul className={styles.diagnosisList}>
+          {searchResults.map((diagnosis, index) => (
+            <li
+              role="menuitem"
+              className={styles.diagnosis}
+              key={index}
+              onClick={() => handleAddDiagnosis(diagnosis)}
+            >
+              {diagnosis.concept.preferredName}
+            </li>
+          ))}
+        </ul>
+      );
+    }
+    return (
+      <Tile light className={styles.emptyResultsText}>
+        <span>
+          {t(
+            "noMatchingDiagnosesText",
+            "No matching diagnoses have been found"
+          )}
+        </span>
+      </Tile>
+    );
+  };
+
+  interface DiagnosisSearchResultsProps {
+    results: Array<Diagnosis>;
+  }
+
+  const DiagnosisSearchResults: React.FC<DiagnosisSearchResultsProps> = ({
+    results
+  }) => <>{results ? <RenderSearchResults /> : <SearchSkeleton />}</>;
```

Comment:
Let's not create components in the body of other components. This can just me moved above or below (but out of) VisitNotesForm.